### ISSUE TITLE
Creating population from a .csv file

### ISSUE THREAD

**Issue by MariaLucia813** [matsim-org/matsim-code-examples#200]
Hello! I am new to the MATSim community and have found it difficult to solve a problem from my case study. I have a .csv that contains id, time and location of several people, however some of these people have more than one location, ie I have lines of the same id only at different times and positions. I have been trying to build my population from this .csv for weeks and have failed to do so. Does anyone have any idea how I could do the population considering that only a few people will have various activities?

**Comment by mrieser** [comment]
What does a single line in the cvs represent? The home-location of persons, or general activity locations of persons?

You mention that for some ids, multiple lines exist. This suggests that a line represents a general activity location. In this case, you could create agent plans based on each activity location for each agent id. Having ids which contain only a single line might be correct, as there is often a small percentage of persons not leaving home during a day (e.g. people being sick, elderly people, having a lazy sunday, ...). In this case, just give them a single home activity, or ignore them completely.

If the share of "single-activity-agents" is too high, then it seems as some data is missing. In this case, assuming you don't have other data, try to make something up. E.g. send them to work or school based on age group, randomly sending some to shopping or leisure afterwards, and then send them back home.

> I have been trying to build my population from this .csv for weeks and have failed to do so. 

What did you try and how did you fail?
It's difficult to help based on this very general note as we don't know what you already tried.

**Comment by OrranaLhaynher** [comment]
I had tried to create the activities based on csv file ids, but failed. However, after several attempts and failures I finally succeeded.
To create the csv file we collected data from an island, and there were lines with repeated ids, but with different days and locations. So I was having a hard time creating the population using csv ids, but I did it.
Thank you very much for your answer! I may have been able to solve it, but I'm very grateful that you answered me.
The scripts to solve this problem are in my git if anyone needs it.

**Comment by MariaLucia813** [comment]
I had tried to create the activities based on csv file ids, but failed. However, after several attempts and failures I finally succeeded.
To create the csv file we collected data from an island, and there were lines with repeated ids, but with different days and locations. So I was having a hard time creating the population using csv ids, but I did it.
Thank you very much for your answer! I may have been able to solve it, but I'm very grateful that you answered me.
The script to solve this problem -> https://github.com/MariaLucia813/matsim-code-examples/blob/11.x/src/main/java/org/matsim/codeexamples/population/CreatingPopulationCSV

After that is just adequate them to your population.
